### DISCUSSION TITLE
Including tags metadata within the artifact pull metrics

### DISCUSSION THREAD

**Discussion by jeroenrinzema** [goharbor/harbor#19705] [Ideas]
Hi, I was wondering if it is possible to include the tags metadata within the artifacts pull metrics allowing us to see on a more fine grained level which artifacts and which tags of a single artifact are pulled.

I went through the code but it was not immediately obvious to me where the metrics are being created.

**Comment by stonezdj** [comment]
In a typical pull request in containerd, it issue the HEAD request to get the tag's digest, then it will pull the image with digest, then the relationship between the pulled digest and its tag are lost, so we could not track the pull metrics of a single tag, but we could track the metrics for an artifact with digest.
